### PR TITLE
Add support for laravel 10

### LINKED ISSUE #222: Support Laravel 10
### Describe the feature

Laravel 10 has released today, I would like to use it with this package

### Use Case

Laravel 10 + aws/aws-sdk-php :)

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### SDK version used

latest

### Environment details (OS name and version, etc.)

-

### PR DESCRIPTION
**Pull request by mateusjunges**
Add support for Laravel 10
Closes #222 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -30,7 +30,10 @@
     "autoload-dev": {
         "psr-4": { "Aws\\Laravel\\Test\\": "tests/" }
     },
-    "extra": {
+  "config": {
+    "allow-plugins": true
```
Comment (SamRemis) [inline composer.json:34]: Why is this being set to true here?

Comment (mateusjunges) [reply composer.json:34]: @SamRemis test suite breaks without it 

https://github.com/mateusjunges/aws-sdk-php-laravel/actions/runs/4177961439

Comment (SamRemis) [reply composer.json:34]: We should instead either allow only the required plugins, or ideally change the test file to use --no-plugins on the composer require statement that's causing that error to be thrown if that is an option.  

I'll spend some time tomorrow looking in to which plugin is causing this and if it's actually required for this test suite

Comment (mateusjunges) [reply composer.json]: Hey @SamRemis I looked into this and indeed this plugin is not necessary for the test suite. I reverted https://github.com/aws/aws-sdk-php-laravel/pull/223/commits/943174f4ea223a18142d8de34920776d065cd196 and added `--no-plugins` when installing composer dependencies.

**Comment by j-fulbright** [comment]
Anxiously awaiting this, so we can migrate our application

**Review by stobrien89 — APPROVED** [review]
LGTM

**Review by SamRemis — APPROVED** [review]

**Comment by stobrien89** [comment]
@mateusjunges,

Thanks for the contribution!  Working on getting our CI updated and then I'll be cutting a release.

**Comment by stobrien89** [comment]
Just cut release 3.8.0.  Thanks all!

**Pull request merged**